src/main/scala/backgammon/Hash.scala

```diff
@@ -0,0 +1,894 @@
+package strategygames.backgammon
+
+import strategygames.Player
+
+final class Hash(size: Int) {
+
+  def apply(situation: Situation): PositionHash = {
+    val l = Hash.get(situation, Hash.polyglotTable)
+    if (size <= 8) {
+      Array.tabulate(size)(i => (l >>> ((7 - i) * 8)).toByte)
+    } else {
+      val m = Hash.get(situation, Hash.randomTable)
+      Array.tabulate(size)(i =>
+        if (i < 8) (l >>> ((7 - i) * 8)).toByte
+        else (m >>> ((15 - i) * 8)).toByte
+      )
+    }
+  }
+}
+
+object Hash {
+
+  val size = 3
+
+  class ZobristConstants(start: Int) {
+    def hexToLong(s: String): Long =
+      (java.lang.Long.parseLong(s.substring(start, start + 8), 16) << 32) |
+        java.lang.Long.parseLong(s.substring(start + 8, start + 16), 16)
+    val p1TurnMask                 = hexToLong(ZobristTables.p1TurnMask)
+    val actorMasks                 = ZobristTables.actorMasks.map(hexToLong)
+    val pocketMasks                = ZobristTables.pocketMasks.map(hexToLong)
+  }
+
+  object ZobristConstants {}
+
+  // The following masks are compatible with the Polyglot
+  // opening book format.
+  private val polyglotTable    = new ZobristConstants(0)
+  private lazy val randomTable = new ZobristConstants(16)
+
+  // for backgammon hashInt is just 1 for stone
+  private def pieceIndex(piece: Piece, count: Int) =
+    (piece.role.hashInt * count) * 2 + piece.player.fold(1, 0)
```

Comment:
you could start at *(count - 1) so that you reduce number of hashes required

Reply:
Probably. I'm wary of potentially multiplying by 0 but you are probably right